homeassistant/components/plugwise/select.py

```diff
@@ -0,0 +1,57 @@
+"""Plugwise Select component for Home Assistant."""
+from __future__ import annotations
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, THERMOSTAT_CLASSES
+from .coordinator import PlugwiseDataUpdateCoordinator
+from .entity import PlugwiseEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Smile selector from a config entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        PlugwiseSelectEntity(coordinator, device_id)
+        for device_id, device in coordinator.data.devices.items()
+        if device["class"] in THERMOSTAT_CLASSES
+        and len(device.get("available_schedules")) > 1
```

Comment:
`len(None)` would be an error

```suggestion
        and len(device.get("available_schedules", [])) > 1
```